packages/home_widget/android/src/main/kotlin/es/antonborri/home_widget/HomeWidgetBackgroundReceiver.kt

```diff
@@ -3,13 +3,34 @@ package es.antonborri.home_widget
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import androidx.work.Data
+import androidx.work.ExistingWorkPolicy
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
 import io.flutter.FlutterInjector
 
 class HomeWidgetBackgroundReceiver : BroadcastReceiver() {
   override fun onReceive(context: Context, intent: Intent) {
     val flutterLoader = FlutterInjector.instance().flutterLoader()
     flutterLoader.startInitialization(context)
     flutterLoader.ensureInitializationComplete(context, null)
-    HomeWidgetBackgroundService.enqueueWork(context, intent)
+    
+    val data = Data.Builder()
+      .putString(
+        HomeWidgetBackgroundWorker.DATA_KEY,
+        intent.data?.toString() ?: ""
+      )
+      .build()
+    
+    val workRequest = OneTimeWorkRequestBuilder<HomeWidgetBackgroundWorker>()
+      .setInputData(data)
+      .build()
+    
+    WorkManager.getInstance(context)
+      .enqueueUniqueWork(
+        HomeWidgetBackgroundWorker.UNIQUE_WORK_NAME,
+        ExistingWorkPolicy.APPEND,
+        workRequest
```

Comment:
I think the `HomeWidgetBackgroundReceiver` would be a little bit cleaner if this would be moved to a 

```kotlin
HomeWidgetBackgroundWorker.enqueWork(context, intent)
```

method similar to the previous BackgroundService